VERSIONING.md

```diff
@@ -135,3 +162,10 @@ on its intended usage (e.g. when calling a method or when passed to any method o
     * Contrib modules will be kept up to date with this project's releases.
 * GitHub releases will be made for all releases.
 * Go modules will be made available at Go package mirrors.
+
+### Long term support
+
+The OpenTelemetry Collector SIG provides long term support for Go modules. Support for modules depend on the module's [target audiences](CONTRIBUTING.md#target-audiences):
+
+- Modules intended for **component developers** MUST be supported for a minimum of **1 year** after the release of the next major version of said module or after the module has been marked as deprecated.
+- Modules intended for **Collector library users** MUST be supported for a minimum of **6 months** after the release of the next major version of said module or after the module has been marked as deprecated.
```

Comment:
I feel the other way :smile: at least for modules intended for component developers. We can't have an ecosystem of components outside of contrib if we don't guarantee that the APIs used to build components are stable and available for a long time. Originally I was going to write 2-3 years for component developers, I went with this to start with something a bit more conservative

For the ones meant for library users, I guess 3 months may be fine now because every major user in that audience (vendors and Jaeger) is involved with the Collector

Reply:
Well I definitely don't want to support the APIs for 3 years, so I can live with 1.